include/mbedtls/md_internal.h

```diff
@@ -79,7 +79,9 @@ extern const mbedtls_md_info_t mbedtls_sha224_info;
 extern const mbedtls_md_info_t mbedtls_sha256_info;
 #endif
 #if defined(MBEDTLS_SHA512_C)
+#if !defined(MBEDTLS_SHA512_NO_SHA384)
 extern const mbedtls_md_info_t mbedtls_sha384_info;
+#endif
```

Comment:
There is many places when you use this sentence with double negation "`#if !defined(MBEDTLS_SHA512_NO_SHA384)`" an it makes riding it is not very friendly. This is a second advantage to use option like `MBEDTLS_SHA512_WITH_SHA384`